matchingengine/OrderBook_test.go

```diff
@@ -53,3 +53,197 @@ func TestLoadOldAsk(t *testing.T) {
+
+	ob.LoadOldBid(limitBid)
+
+}
+
+func TestLoadOldBidStopLoss(t *testing.T) {
+
+	// This initializes the logger. This is required because LoadOldAsk tries accessing the logger
+	// and if it is not initialized, a nil reference error is thrown.
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl := gomock.NewController(t)
+	defer mockControl.Finish()
+
+	var stockID uint32 = 1
+	var stockQuantity uint32 = 10
+	var stockPrice uint32 = 20
+
+	mockAskQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidQueue := mocks.NewMockBidPQueue(mockControl)
+	mockAskStoplossQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidStoplossQueue := mocks.NewMockBidPQueue(mockControl)
+	mockDepth := mocks.NewMockMarketDepthStream(mockControl)
+
+	ob := &orderBook{
+		logger: utils.Logger.WithFields(logrus.Fields{
+			"module":        "matchingengine.OrderBook.test",
+			"param_stockId": stockID,
+		}),
+		stockId:     stockID,
+		askChan:     make(chan *models.Ask),
+		bidChan:     make(chan *models.Bid),
+		asks:        mockAskQueue,
+		bids:        mockBidQueue,
+		askStoploss: mockAskStoplossQueue,
+		bidStoploss: mockBidStoplossQueue,
+		depth:       mockDepth,
+	}
+
+	stopLossBid := makeBid(1, stockID, models.StopLoss, stockQuantity, stockPrice, "")
+
+	mockBidStoplossQueue.EXPECT().Push(stopLossBid).Times(1)
+
+	ob.LoadOldBid(stopLossBid)
+
+}
+
+func TestLoadOldTransactions(t *testing.T) {
+
+	// This initializes the logger. This is required because LoadOldAsk tries accessing the logger
+	// and if it is not initialized, a nil reference error is thrown.
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl := gomock.NewController(t)
+	defer mockControl.Finish()
+
+	var stockID uint32 = 1
+
+	mockAskQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidQueue := mocks.NewMockBidPQueue(mockControl)
+	mockAskStoplossQueue := mocks.NewMockAskPQueue(mockControl)
+	mockBidStoplossQueue := mocks.NewMockBidPQueue(mockControl)
+	mockDepth := mocks.NewMockMarketDepthStream(mockControl)
+
+	ob := &orderBook{
+		logger: utils.Logger.WithFields(logrus.Fields{
+			"module":        "matchingengine.OrderBook.test",
+			"param_stockId": stockID,
+		}),
+		stockId:     stockID,
+		askChan:     make(chan *models.Ask),
+		bidChan:     make(chan *models.Bid),
+		asks:        mockAskQueue,
+		bids:        mockBidQueue,
+		askStoploss: mockAskStoplossQueue,
+		bidStoploss: mockBidStoplossQueue,
+		depth:       mockDepth,
+	}
+
+	t1 := &models.Transaction{
+		Id:            1,
+		UserId:        10,
+		StockId:       stockID,
+		Type:          1,
+		StockQuantity: 20,
+		Price:         50,
+		Total:         1000,
+		CreatedAt:     "random1",
+	}
+
+	t2 := &models.Transaction{
+		Id:            2,
+		UserId:        10,
+		StockId:       stockID,
+		Type:          2,
+		StockQuantity: 10,
+		Price:         20,
+		Total:         200,
+		CreatedAt:     "random2",
+	}
+
```

Comment:
Can we have a helper function `makeTransaction` just like `makeAsk`/`makeBid`?

Reply:
Done for Transaction. For now, let's create order books the way we're doing now.